chromadb/api/models/Collection.py

```diff
@@ -150,9 +153,30 @@ def add(
         # We need to compute the embeddings if they're not provided
         if embeddings is None:
             # At this point, we know that one of documents or images are provided from the validation above
+
+            def chunk_and_assign_ids(
+                input: Embeddable, ids: IDs, chunker: Chunker[Embeddable]
+            ) -> Tuple[List[Embeddable], List[IDs]]:
+                chunks = chunker(input)
+                chunk_ids = []
+                for chunk, id in zip(chunks, ids):
+                    chunk_ids.append([f"{id}_{i}" for i in range(len(chunk))])
+                return chunks, chunk_ids
+
             if documents is not None:
+                if chunker is not None:
+                    chunks, chunk_ids = chunk_and_assign_ids(documents, ids, chunker)
+                    # Flatten chunk_ids and chunks, and assign them as the new values
+                    ids = [id for chunk_id in chunk_ids for id in chunk_id]
```

Comment:
Basically we expand the list of IDs by chunk ID. We should probably pass these out to the user (we currently do not pass IDs out) so they can keep track of the new IDs. I can do that here, on in a separate PR.